common/lib/protocol-definitions/src/clients.ts

```diff
@@ -5,27 +5,77 @@
 
 import { IUser } from "./users";
 
+/**
+ * A client's connection mode. Readonly or read/write.
+ */
 export type ConnectionMode = "write" | "read";
```

Comment:
nitpick: doc refers to read-only or read AND write but the strings used for the mode's themselves are either "write" or "read"

Reply:
read-only is not good here, IMHO. It's too late to change it, but I wish it was "view-only", not "read". view-only, in a sense that it does not mean user has no permissions to change file. Also no matter if user has write permissions, in view-only mode client could send signals.
Not sure if my description helps :) The good news - this is not exposed outside of protocol, mostly.

Reply:
Thanks @vladsud I'll update the description accordingly. Even if we're stuck with the flag naming, we can at least try to clarify meaning in the docs here 🙂

Reply:
I took a stab at capturing your notes in a descriptive comment. Let me know if you think it could use further clarification.